iib/workers/tasks/build.py

```diff
@@ -784,6 +931,26 @@ def handle_add_request(
     _update_index_image_build_state(request_id, prebuild_info)
 
     with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
+        if from_index:
+            msg = 'Checking if bundles are already present in index image'
+            log.info(msg)
+            set_request_state(request_id, 'in_progress', msg)
+            present_bundles = _get_present_bundles(from_index, temp_dir)
+            resolved_bundles = _get_missing_bundles(present_bundles, resolved_bundles)
+            if not resolved_bundles:
+                log.info(
+                    'All bundles are already present in the index image.'
+                    ' No additional operations are necessary'
+                )
+                set_request_state(
+                    request_id, 'complete', 'All bundles are already present in the index image'
+                )
+                return
+            log.info(
```

Comment:
It might be more useful to print the ones which are already present and suggest that they will not be added again.

Reply:
I'm thinking if we can also add a state_reason update with the same info as above. thoughts?

Reply:
I have switched the log message. But I don't think it makes much sense to put it as a state

Reply:
We can add it later if stakeholders need it.